### PR TITLE
Handle handshake failure in Websocket Client example

### PR DESCRIPTION
**Pull request by MartyIX**
Pull request for #6998

### CODE REVIEW

**Comment by MartyIX** [comment]
Hm. I don't really know how to find out what style checking rule is failing.

**Comment by normanmaurer** [comment]
@MartyIX https://garage.netty.io/teamcity/viewLog.html?buildId=22423&buildTypeId=pulls_netty&tab=buildLog#_focus=4316 ... Also you can just run `mvn clean package` and it should fail for you as well.

**Comment by normanmaurer** [comment]
@MartyIX also please sign our ICLA: https://netty.io/s/icla

**Comment by MartyIX** [comment]
Ok. I've just signed.

File: example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java [hunk 1]
```diff
@@ -82,9 +83,14 @@ public void channelInactive(ChannelHandlerContext ctx) {
     public void channelRead0(ChannelHandlerContext ctx, Object msg) throws Exception {
         Channel ch = ctx.channel();
         if (!handshaker.isHandshakeComplete()) {
-            handshaker.finishHandshake(ch, (FullHttpResponse) msg);
-            System.out.println("WebSocket Client connected!");
-            handshakeFuture.setSuccess();
+            try {
+                handshaker.finishHandshake(ch, (FullHttpResponse) msg);
+                System.out.println("WebSocket Client connected!");
```
Comment (netkins) [inline example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS106)

File: example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java [hunk 2]
```diff
@@ -82,9 +83,14 @@ public void channelInactive(ChannelHandlerContext ctx) {
...
+                handshakeFuture.setSuccess();
+            } catch (WebSocketHandshakeException e) {
+                System.out.println("WebSocket Client failed to connect");
```
Comment (netkins) [inline example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java]: MAJOR Replace this usage of System.out or System.err by a logger. [rule](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS106)

File: example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java (hunk 1 above)
Comment (MartyIX) [inline example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java]: @normanmaurer Is it necessary to replace it? The code is an example. It's easier to grab code with println and run it somewhere without configuring of a logger..

File: example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java (hunk 2 above)
Comment (normanmaurer) [inline example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java]: I think its fine as we also use println for the success case

Comment (MartyIX) [inline example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java]: @normanmaurer OK. So what should I do next to get this pull request merged?

Comment (normanmaurer) [inline example/src/main/java/io/netty/example/http/websocketx/client/WebSocketClientHandler.java]: nothing... I will merge soon.

**Comment by normanmaurer** [comment]
Cherry-picked into 4.1 (f897507b09d61baca231f43bfc701c134966bd11) and 4.0 (58dc0b29752b46be978648cef6151f41a9dbaf52)

**Comment by normanmaurer** [comment]
@MartyIX thanks!

**Pull request closed without merging**